utxonursery.go

```diff
@@ -346,6 +352,13 @@ func (u *utxoNursery) IncubateOutputs(chanPoint wire.OutPoint,
 	incomingHtlcs []lnwallet.IncomingHtlcResolution,
 	broadcastHeight uint32) error {
 
+	// Add to wait group because nursery might shut down during execution of
+	// this function. Otherwise it could happen that nursery thinks it is
+	// shut down, but in this function new goroutines were started and stay
+	// around.
+	u.wg.Add(1)
```

Comment:
Another potential avenue for trying to ensure that we don't shut down too early might be to lock around the closing of `quit` in the stop method. inside the critical section below, we can then check the quit chan to see if we should bail early. I think this may be less racey wrt to modifying the waitgroup counter too late

Reply:
Added commit that checks for quit channel. I think it is just as safe as checking inside the critical section, but needs less code.